src/Symfony/Component/Routing/Loader/Configurator/CollectionConfigurator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader\Configurator;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class CollectionConfigurator
+{
+    use Traits\AddTrait;
+    use Traits\RouteTrait;
```

Comment:
shouldn't `->defaults`, `->requirements` and so on apply on `$this->collection` in this class ?

Reply:
That's the case, but not on the collection yes, because it works reverse: calling on the collection means changing routes declared before. Here, we want first to configure the route *prototype* (`$this->route` in this class), then use this prototype as a template for new routes added.
The goal is to make this work as expected:
```
$collection->defaults(['foo' => 'bar'])->add(...);
```

but I'm not sure doing `$collection->add(...);` *then* `$collection->defaults(['foo' => 'bar']);` and having it applied to previously declared routes in the collection is something we want to have.